src/activity_handlers.cpp

```diff
@@ -631,30 +920,131 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
+        g->m.add_splatter( type_blood, p->pos(), rng( corpse->size + 2, ( corpse->size + 1 ) * 2 ) );
+        for( int i = 1; i <= corpse->size; i++ ) {
+            g->m.add_splatter_trail( type_gib, p->pos(), random_entry( g->m.points_in_radius( p->pos(), corpse->size + 1 ) ) );
+            g->m.add_splatter_trail( type_blood, p->pos(), random_entry( g->m.points_in_radius( p->pos(), corpse->size + 1 ) ) );
+        }
+        act->set_to_null();
+        return;
+        }
     }
 
-    // Recover hidden items
-    for( auto &content : contents  ) {
-        if( ( roll_butchery() + 10 ) * 5 > rng( 0, 100 ) ) {
-            //~ %1$s - item name, %2$s - monster name
-            p->add_msg_if_player( m_good, _( "You discover a %1$s in the %2$s!" ), content.tname().c_str(),
-                     corpse->nname().c_str() );
-            g->m.add_item_or_charges( p->pos(), content );
-        } else if( content.is_bionic()  ) {
-            g->m.spawn_item(p->pos(), "burnt_out_bionic", 1, 0, age);
+    // all action types - yields
+    if( corpse->harvest.is_null() ) {
+        butchery_drops_hardcoded( &corpse_item, corpse, p, age, roll_butchery, action );
+    } else {
+        butchery_drops_harvest( &corpse_item, *corpse, *p, age, roll_butchery, action );
+    }
+
+    // quick & full butcher only
+    // reveal hidden items / hidden content
+    if( action != F_DRESS ) {
+        for( auto &content : contents  ) {
+            if( ( roll_butchery() + 10 ) * 5 > rng( 0, 100 ) ) {
+                //~ %1$s - item name, %2$s - monster name
+                p->add_msg_if_player( m_good, _( "You discover a %1$s in the %2$s!" ), content.tname().c_str(),
+                        corpse->nname().c_str() );
+                g->m.add_item_or_charges( p->pos(), content );
+            } else if( content.is_bionic()  ) {
+                g->m.spawn_item(p->pos(), "burnt_out_bionic", 1, 0, age);
+            }
         }
     }
 
-    p->add_msg_if_player( m_good, _("You finish butchering the %s."), corpse->nname().c_str() );
+    //end messages and effects
+    switch( action ) {
+    case QUARTER:
+        break;
+    case BUTCHER:
+        p->add_msg_if_player( m_good, _("After quick attempt to harvest flesh from %s you leave the rest for the vultures."), corpse_item.tname().c_str() );
+        g->m.i_rem( p->pos(), act->index );
+        break; //no set_to_null here, for multibutchering
+    case BUTCHER_FULL:
+        p->add_msg_if_player( m_good, _("You finish butchering the %s."), corpse_item.tname().c_str() );
+        g->m.i_rem( p->pos(), act->index );
+        act->set_to_null();
+        return;
+        break;
+    case F_DRESS:
+        if ( roll_butchery() < 0 ) { // partial failure
+            switch( rng( 1, 3 ) ) {
+                case 1:
+                    p->add_msg_if_player(m_warning, _( "You unskilfully hack the corpse and chop off some excess body parts.  You think this was not a job well done." ) );
```

Comment:
Correction: `unskillfully`
Suggestion: `You unskillfully hack up the corpse and chop off some excess body parts.  You're left wondering how you did so poorly.`